### PR TITLE
Add frontend for legacy api/irc key

### PR DESCRIPTION
**Pull request by nanaya**
The wordings may need some fixing.

Depends on:
- [x] #10123
- [x] #10117

### CODE REVIEW

File: resources/js/legacy-api-key/controller.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  showing_form: boolean;
+}
+
+type DatasetState = Partial<State> & Required<Pick<State, 'legacy_api_key'>>;
+
+export default class Controller {
+  @observable state;
+  private stateSyncDisposer;
+  private xhrCreate?: JQuery.jqXHR<LegacyApiKeyJson>;
+  private xhrDelete?: JQuery.jqXHR<void>;
```
Comment (notbakaneko) [inline resources/js/legacy-api-key/controller.ts:20]: needs `@observable`

File: resources/js/legacy-irc-key/index.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    this.controller.destroy();
+  }
+
+  render() {
+    return this.controller.key == null ? this.renderEmpty() : this.renderDetails();
+  }
+
+  @action
+  private readonly onNewKeyClick = () => {
+    if (!confirm('legacy_irc_key.confirm_new')) return;
```
Comment (notbakaneko) [inline resources/js/legacy-irc-key/index.tsx:38]:
```suggestion
    if (!confirm(trans('legacy_irc_key.confirm_new'))) return;
```

**Review by notbakaneko — CHANGES_REQUESTED** [review]
I've just realized this should be `app_url` not `$field` (and the also maybe the `foreach`?) https://github.com/ppy/osu-web/blob/master/app/Models/ApiKey.php#L67

File: resources/js/legacy-api-key/form.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  private errors = new FormErrors();
+
+  constructor(props: Props) {
+    super(props);
+
+    makeObservable(this);
+  }
+
+  @action
+  readonly onCancelClick = () => {
```
Comment (notbakaneko) [inline resources/js/legacy-api-key/form.tsx:31]: `readonly`

File: resources/js/legacy-api-key/details.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  private readonly deleteClicked = () => {
+    if (!confirm(trans('common.confirmation'))) return;
+
+    this.props.controller.deleteKey();
+  };
+
+  @action
+  private onClickToggleKeyVisibility = () => {
```
Comment (notbakaneko) [inline resources/js/legacy-api-key/details.tsx:115]: `readonly`

File: resources/js/legacy-irc-key/controller.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    makeObservable(this);
+
+    this.stateSyncDisposer = reaction(
+      () => JSON.stringify(this.state),
+      (stateString) => this.container.dataset.state = stateString,
+    );
+  }
+
+  createKey() {
```
Comment (notbakaneko) [inline resources/js/legacy-irc-key/controller.ts:42]: also needs `@action` to go with the `@observable` to properly show the spinner or disable button, same for the other create/delete keys

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: resources/js/legacy-api-key/form.tsx (hunk 3 above)
Comment (nanaya) [reply resources/js/legacy-api-key/form.tsx:31]: but it's already readonly 🤔

**Review by notbakaneko — CHANGES_REQUESTED** [review]
can also do with `attributes` key in model_validation to localise key name
image

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**